Adding release notes automation to the IG

coxstuart opened the pull request:
Adding release notes automation

- [x] Add automated release notes and history.html page generation when release is created
- [x] Add release notes link to readme
- [x] Fix broken xml examples link on download page
- [x] Remove version from canonical link

dmangum2 commented on README.md:
```diff
@@ -39,8 +39,4 @@ IBM and the IBM logo are trademarks of International Business Machines Corporati
-| Date           | IG Version      | FHIR Version | Description |
-| -------------- | --------------- | ------------ | ----------- |
-|  7/23/2021    | [0.1.1](https://alvearie.github.io/alvearie-fhir-ig/)           | 4.0.1        | Release       |
+View the [Release][https://github.com/Alvearie/alvearie-fhir-ig/releases]. Each release includes the full IG specication. 
```
"specication" typo.

dmangum2 replied:
Also on the readme.md, looks strange.  (see Preview tab) ![image](https://user-images.githubusercontent.com/85960782/127879926-f52b7a52-811b-40a3-999b-886f16a33d21.png)

coxstuart replied:
Thank you.  Fixed the typos and added some more content.

dmangum2 reviewed (changes requested):
One change requested.

coxstuart commented:
> Issue labels "will" create? Or "are created on ...."

@dmangum2:  Thank you!  I fixed multiple typos and rewrote a bunch of the readme.  Please take another look.  I need "gramerly for git". Do they make that?

dmangum2 reviewed (approved):
Looks good!

kimkenna commented:
Two typos in the readme I believe:
1. Line 37 is missing an 's' for the word Issues in this line: "* Issue are used to document changes included in the release notes for each release"
2. Line 55 is missing the preceding e for export in this line: "xport GREN_GITHUB_TOKEN=<YOUR TOKEN HERE>"

coxstuart commented:
> Two typos in the readme I believe:
> 
> 1. Line 37 is missing an 's' for the word Issues in this line: "* Issue are used to document changes included in the release notes for each release"
> 2. Line 55 is missing the preceding e for export in this line: "xport GREN_GITHUB_TOKEN="

Thanks @kimkenna.  Sharp eyes!  Fixed the readme.

dmangum2 commented on README.md:
```diff
@@ -39,8 +94,43 @@ IBM and the IBM logo are trademarks of International Business Machines Corporati
-| Date           | IG Version      | FHIR Version | Description |
-| -------------- | --------------- | ------------ | ----------- |
-|  7/23/2021    | [0.1.1](https://alvearie.github.io/alvearie-fhir-ig/)           | 4.0.1        | Release       |
+Changes must be made to the following files to version the IG. This will ensure the automatmed releases work correctly.
```
automatmed misspelled

coxstuart replied:
Fixed. Thank you.

dmangum2 commented on README.md:
```diff
@@ -39,8 +94,43 @@ IBM and the IBM logo are trademarks of International Business Machines Corporati
+
+IG releases and version numbers should be modified as follows:
+
+1. **New major versions on breaking changes**
```
Should we note the version number scheme as N.x.x, so it's clear what pattern we are going to use?

coxstuart replied:
Good suggestion.  I added that.  Please review and let me know if you think we need more.

dmangum2 commented on README.md:
```diff
@@ -39,8 +94,43 @@ IBM and the IBM logo are trademarks of International Business Machines Corporati
+    - Code changes to existing vocabularies
+    - Url changes to existing profiles, vocabularies, etc.
+
+2. **New minor version on non-breaking IG profile changes**
```
Note the version number pattern, like x.N.x?

coxstuart replied:
Done

dmangum2 commented on README.md:
```diff
@@ -39,8 +94,43 @@ IBM and the IBM logo are trademarks of International Business Machines Corporati
+    - Looser data type or valueset binding constraints on existing profiles
+    - New optional fields on existing profiles
+  
+3. **New revision on documentation changes that are reflected in the IG**
```
Note the version number pattern, like x.x.N?

coxstuart replied:
Done

dmangum2 reviewed (changes requested):
I had a few comments.

kimkenna reviewed (approved):
Great work!
